test/test-source-map-generator.js

```diff
@@ -252,6 +252,24 @@ exports['test .fromSourceMap with sourcesContent'] = function (assert) {
   util.assertEqualMaps(assert, map.toJSON(), util.testMapWithSourcesContent);
 };
 
+exports['test .fromSourceMap with single source'] = function (assert) {
+  var map = SourceMapGenerator.fromSourceMap(
+      new SourceMapConsumer(util.testMapSingleSource));
+  util.assertEqualMaps(assert, map.toJSON(), util.testMapSingleSource);
+};
+
+exports['test .fromSourceMap with empty mappings'] = function (assert) {
+  var map = SourceMapGenerator.fromSourceMap(
+    new SourceMapConsumer(util.testMapEmptyMappings));
+  util.assertEqualMaps(assert, map.toJSON(), util.testMapEmptyMappings);
+};
+
+exports['test .fromSourceMap with empty mappings and relative sources'] = function (assert) {
+  var map = SourceMapGenerator.fromSourceMap(
+    new SourceMapConsumer(util.testMapEmptyMappingsRelativeSources));
+  util.assertEqualMaps(assert, map.toJSON(), util.testMapEmptyMappingsRelativeSources_generatedExpected);
+};
+
```

Comment:
We also need a test for the following scenario:
- We have two sources: foo.js and bar.js
- The mappings are NOT empty, but all the mappings refer only to foo.js.
- The test should assert that bar.js is also included in the list of sources.

Note that this test will fail with your current code, because it will only add bar.js if the mappings are empty.